api/v1alpha1/ingressnodefirewall_types.go

```diff
@@ -76,71 +100,68 @@ type IngressNodeFirewallProtocolRule struct {
 type IngressNodeFirewallRuleProtocolType string
 
 const (
-	// ProtocolTypeICMP refers to the ICMP protocol
-	ProtocolTypeICMP IngressNodeFirewallRuleProtocolType = "icmp"
+	// ProtocolTypeICMP refers to the ICMP protocol.
+	ProtocolTypeICMP IngressNodeFirewallRuleProtocolType = "ICMP"
 
-	// ProtocolTypeICMP6 refers to the ICMPv6 protocol
-	ProtocolTypeICMP6 IngressNodeFirewallRuleProtocolType = "icmp6"
+	// ProtocolTypeICMP6 refers to the ICMPv6 protocol.
+	ProtocolTypeICMP6 IngressNodeFirewallRuleProtocolType = "ICMPv6"
 
-	// ProtocolTypeTCP refers to the TCP protocol, for either IPv4 or IPv6
-	ProtocolTypeTCP IngressNodeFirewallRuleProtocolType = "tcp"
+	// ProtocolTypeTCP refers to the TCP protocol, for either IPv4 or IPv6.
+	ProtocolTypeTCP IngressNodeFirewallRuleProtocolType = "TCP"
 
-	// ProtocolTypeUDP refers to the UDP protocol, for either IPv4 or IPv6
-	ProtocolTypeUDP IngressNodeFirewallRuleProtocolType = "udp"
+	// ProtocolTypeUDP refers to the UDP protocol, for either IPv4 or IPv6.
+	ProtocolTypeUDP IngressNodeFirewallRuleProtocolType = "UDP"
 
-	// ProtocolTypeSCTP refers to the SCTP protocol, for either IPv4 or IPv6
-	ProtocolTypeSCTP IngressNodeFirewallRuleProtocolType = "sctp"
+	// ProtocolTypeSCTP refers to the SCTP protocol, for either IPv4 or IPv6.
+	ProtocolTypeSCTP IngressNodeFirewallRuleProtocolType = "SCTP"
 )
 
-// IngressNodeFirewallActionType indicates whether an IngressNodeFirewallRule allows or denies traffic
-// +kubebuilder:validation:Pattern=`^allow|deny$`
+// IngressNodeFirewallActionType indicates whether an IngressNodeFirewallRule allows or denies traffic.
+// +kubebuilder:validation:Enum="Allow";"Deny"
 type IngressNodeFirewallActionType string
 
 const (
-	IngressNodeFirewallAllow IngressNodeFirewallActionType = "allow"
-	IngressNodeFirewallDeny  IngressNodeFirewallActionType = "deny"
+	IngressNodeFirewallAllow IngressNodeFirewallActionType = "Allow"
+	IngressNodeFirewallDeny  IngressNodeFirewallActionType = "Deny"
 )
 
-// IngressNodeFirewallRules define ingress node firewall rule
+// IngressNodeFirewallRules define ingress node firewall rule.
 type IngressNodeFirewallRules struct {
-	// SourceCIDRS is A list of CIDRs from which we apply node firewall rule
-	// +kubebuilder:validation:one-of[]=Type=string;Format:cidr
```

Comment:
did you try and get `cidr` working?

Reply:
no I thought I did remove it from the last diffs ?

Reply:
yeah its removed we can check it later if we can make it work I can get IPv4 to work but not CIDR